Tests/KlaviyoSwiftTests/StateManagementTests.swift

```diff
@@ -0,0 +1,366 @@
+    }
+    
+    func testQueueThatIsFlushingDoesNotFlush() async throws {
+        let apiKey = "fake-key"
+        let initialState = KlaviyoState(apiKey: apiKey,
+                                        queue: [],
+                                        requestsInFlight: [],
+                                        initialized: true,
+                                        flushing: true)
+        let store = TestStore(initialState: initialState, reducer: KlaviyoReducer())
+        _ = await store.send(.flushQueue)
+    }
+    
+    func testEmptyQueueDoesNotFlush() async throws {
+        let apiKey = "fake-key"
+        let initialState = KlaviyoState(apiKey: apiKey,
+                                        queue: [],
+                                        requestsInFlight: [],
+                                        initialized: true,
+                                        flushing: false)
+        let store = TestStore(initialState: initialState, reducer: KlaviyoReducer())
+        _ = await store.send(.flushQueue)
+    }
+    
+    func testSendRequestWithNoRequestsInFlight() async throws {
+        let apiKey = "fake-key"
+        let initialState = KlaviyoState(apiKey: apiKey,
+                                        queue: [],
+                                        requestsInFlight: [],
+                                        initialized: true,
+                                        flushing: true)
+        let store = TestStore(initialState: initialState, reducer: KlaviyoReducer())
+        // Shouldn't really but getting more coverage...
+        _ = await store.send(.sendRequest) {
+            $0.flushing = false
+        }
+    }
+    
+    func testFlushQueueWithMultipleRequests() async throws {
+        let apiKey = "fake-key"
+        var count = 0
+        // request uuids need to be unique :)
+        environment.analytics.uuid = {
+            count += 1
+            switch count {
+            case 1:
+                return UUID(uuidString: "00000000-0000-0000-0000-000000000001")!
+            case 2:
+                return UUID(uuidString: "00000000-0000-0000-0000-000000000002")!
+            default:
+                return UUID(uuidString: "00000000-0000-0000-0000-000000000003")!
+            }
+        }
+        var initialState = KlaviyoState(apiKey: apiKey,
+                                        anonymousId: environment.analytics.uuid().uuidString,
+                                        pushToken: "blob_token",
+                                        queue: [],
+                                        requestsInFlight: [],
+                                        initialized: true,
+                                        flushing: false)
+        let request = try initialState.buildProfileRequest()
+        let request2 = try initialState.buildTokenRequest()
+        initialState.queue = [request, request2]
+        let store = TestStore(initialState: initialState, reducer: KlaviyoReducer())
```

Comment:
These tests walk you through state changes in the store. I have them set for exhaustive mode which means you have to spell out each change that occurs as well every action that is fired.